File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function(nums) {
    let start = 0, end = nums.length-1;
    while(start<end){
        let mid = start + Math.floor((end-start)/2);
        if(mid %2 == 1) mid--;
        if(nums[mid] != nums[mid+1]) end = mid;
        else start = mid+2;
    }
    return nums[end];
};

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int start = 0, end = nums.length-1;
        while(start<end){
            int mid = start + (end - start)/2;
            if(mid %2 == 1){
                mid--;
            }
            if(nums[mid] != nums[mid+1]){
                end = mid;
            }else{
                start = mid+2;
            }
        }
        return nums[end];
    }
}